function-aws/src/main/java/io/micronaut/function/aws/event/AfterExecutionEvent.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.function.aws.event;
+
+import io.micronaut.core.annotation.Nullable;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * This event is published after the execution of {@link io.micronaut.function.aws.MicronautRequestHandler#execute(Object)}
+ * and {@link io.micronaut.function.aws.MicronautRequestStreamHandler#execute(InputStream, OutputStream)} methods to allow
+ * performing actions before the Lambda function run is finished and the JVM is hibernated.
+ * <p>
+ * This event must be processed synchronously to guarantee it has been processed before the Lambda funciton is hibernated.
+ *
+ * @author Vladimir Orany
+ * @since 3.8.0
+ */
+public class AfterExecutionEvent {
+
+    /**
+     * Creates a new {@link AfterExecutionEvent} with an optional result of the execution.
+     * @param output an optional result of the exectuion
+     * @return a new {@link AfterExecutionEvent} with an optional result of the execution
+     */
+    public static AfterExecutionEvent success(@Nullable Object output) {
+        return new AfterExecutionEvent(output, null);
+    }
+
+    /**
+     * Creates a new {@link AfterExecutionEvent} with an exception been thrown.
+     * @param exception the exception which has been thrown during the execution
+     * @return a new {@link AfterExecutionEvent} with an exception been thrown.
+     */
+    public static AfterExecutionEvent failure(RuntimeException exception) {
```

Comment:
Why `RuntimeException`? Just using `Exception` should be ok

Reply:
I initially had `Throwable` but as `execute` method is not throwing any checked exceptions the only `RuntimeException` can happen under normal circumstances. I'm easy to broaden it to `Exception` or `Throwable` if you think it's better.

Reply:
nvm, I have noticed that `MicronautRequestStreamHandler` is actually throwing `IOException`.